dashboard/app/reporting_test.go

```diff
@@ -721,3 +721,62 @@ func TestAltTitles7(t *testing.T) {
 	c.expectEQ(rep.Title, crash1.Title)
 	c.expectEQ(rep.Log, crash2.Log)
 }
+
+func TestDetachExternalTracker(t *testing.T) {
+	c := NewCtx(t)
+	defer c.Close()
+
+	build := testBuild(1)
+	c.client.UploadBuild(build)
+
+	crash1 := &dashapi.Crash{
+		BuildID:     "build1",
+		Title:       "title1",
+		Maintainers: []string{`"Foo Bar" <foo@bar.com>`, `bar@foo.com`},
+		Log:         []byte("log1"),
+		Report:      []byte("report1"),
+		MachineInfo: []byte("machine info 1"),
+	}
+	c.client.ReportCrash(crash1)
+
+	// Get single report for "test" type.
+	resp, _ := c.client.ReportingPollBugs("test")
+	c.expectEQ(len(resp.Reports), 1)
+	rep := resp.Reports[0]
+	c.expectNE(rep.ID, "")
+	c.expectEQ(string(rep.Config), `{"Index":1}`)
+
+	// Signal detach_reporting for current bug.
+	reply, _ := c.client.ReportingUpdate(&dashapi.BugUpdate{
+		ID:         rep.ID,
+		Status:     dashapi.BugStatusUpstream,
+		ReproLevel: dashapi.ReproLevelNone,
+		Link:       "http://URI/1",
+		CrashID:    rep.CrashID,
+	})
+	c.expectEQ(reply.OK, true)
+
+	// Now add syz repro to check it doesn't use first reporting.
+	crash1.ReproOpts = []byte("some opts")
+	crash1.ReproSyz = []byte("getpid()")
+	c.client.ReportCrash(crash1)
+
+	// Fetch bug and check reporting path (Config) is different.
+	resp, _ = c.client.ReportingPollBugs("test")
+	c.expectEQ(len(resp.Reports), 1)
+	rep2 := resp.Reports[0]
+	c.expectNE(rep2.ID, "")
+	c.expectEQ(string(rep2.Config), `{"Index":2}`)
+
+	// No notification happened.
+	c.advanceTime(20 * 24 * time.Hour)
+	c.client.pollNotifs(0)
+
+	// Accept second reporting.
+	reply, _ = c.client.ReportingUpdate(&dashapi.BugUpdate{
```

Comment:
It really should not reply after 20 days :)
`c.client.pollBug()` will acknowledge report with `BugStatusOpen` automatically, so I think we can just drop `c.advanceTime(20 * 24 * time.Hour)` and everything below, since `c.Close()` will do advanceTime and check there are no pending notifs/reports.